Maple2.File.Parser/NifParser.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Maple2.File.IO.Crypto.Common;
+using Maple2.File.IO.Nif;
+using Maple2.File.Parser.Tools;
+
+namespace Maple2.File.Parser;
+
+public class NifParser {
+    private readonly List<NifM2dArchive> modelM2dReaders;
+
+    public NifParser(List<NifM2dArchive> modelM2dReaders) {
+        this.modelM2dReaders = modelM2dReaders;
+    }
+
+    public IEnumerable<(uint Llid, string relpath, List<NiPhysXProp> PhysXProps)> Parse() {
+        foreach (NifM2dArchive nifReader in modelM2dReaders) {
+            foreach (PackFileEntry entry in nifReader.M2dReader.Files.Where(entry => entry.Name.EndsWith(".nif"))) {
+                string path = nifReader.PathPrefix + entry.Name;
+                uint llid = LlidHash.Hash(path);
+
+                NifDocument nifDocument = new NifDocument(path, nifReader.M2dReader.GetBytes(entry));
+
+                try {
+                    nifDocument.Parse();
+                } catch (Exception ex) {
+                    Console.WriteLine("Error reading nif: " + path);
```

Comment:
no logs, use a string builder to build a string and throw it in the exception

Reply:
Is there a way I can append the extra info to the exception without getting rid of the stack trace? That's the main problem those logs were meant to solve

Reply:
yes, just pass the exception into the new exception